accepted/00104-enhanced-ansible-support.md

```diff
@@ -0,0 +1,296 @@
+
+We're currently using the `state module`:
+```
+run_ansible_playbook:
+  ansible.playbooks:
+     - .......
+```
+
+To get the raw output we will have to use module.run to call the `ansible.playbooks` function from the `execution module`:
+```
+run_ansible_playbook:
+  module.run:
+    - name: ansible.playbooks
+    - ....
+```
+
+The changed `runplaybook.sls` would then look like:
+```
+run_ansible_playbook:
+  mgrcompat.module_run:
+    - name: ansible.playbooks
+    - playbook: {{ pillar["playbook_path"] }}
+    - rundir: {{ pillar["rundir"] }}
+    - flush_cache: {{ pillar["flush_cache"] }}
+{%- if "inventory_path" in pillar %}
+    - inventory: {{ pillar["inventory_path"] }}
+{% endif %}
+```
+
+#### Displaying targeted systems
+
+Since the list of systems targeted by an ansible playbook can be very long, I propose to display the used inventory as a link in the event history details instead.
+
+In case the selected event details are of the playbook action type we will set an additional request parameter called `inventory` in `SystemHistoryEventAction.java` and render an extra row that would look like:
+```
+Inventory: <link_to_inventory>
+```
+
+Clicking the link would then forward the user to the inventory file in `System` > `Ansible` > `Inventories`.
+
+The inventory then displays the list of systems registered in Uyuni that were targeted by the playbook execution.
+
+Since pre selecting a specific inventory based on the provided URL is not currently supported by the [ansible-path-content.tsx](https://github.com/uyuni-project/uyuni/blob/master/web/html/src/manager/minion/ansible/ansible-path-content.tsx) component, it will have to be adapted to support it.
+The URL to load a specific inventory would look like:
+```
+manager/systems/details/ansible/inventories?sid=<system_id>&iid=<inventory_id>
+```
+with `<system_id>` being the system id of the ansible control node and
+`<inventory_id` being the id of the inventory to load.
+
+## Add UI support to allow editing of variables defined in the playbook
+
+The plan here is to enhance the existing UI in `System > Ansible > Playbooks > select playbook` with generated input fields based on the `vars` defined in the playbook.
+
+For example
+```
+- name: Example playbook with various variable types
+  hosts: all
+  vars:
+    # List variable
+    software_packages:
+      - git
+      - nginx
+      - curl
+
+    # Dictionary variable
+    user_info:
+      username: "devuser"
+      uid: 1001
+      home_dir: "/home/devuser"
+
+    # Integer variable
+    max_open_files: 1024
+
+    # Boolean variable
+    enable_firewall: true
+
+    # String variable
+    welcome_message: "Welcome to the server setup!"
+```
+The UI would generate input fields based on variable type (list, dict, string) that are pre-populated with the default variables set in the playbook (if any).
```

Comment:
Since we already have a fair bit of code that does many similar things for formulas, do you think we could extend/reuse that, or do you think the underlying requirements are too different?

Reply:
Underlaying requirements aren't that different. If we could reuse some exiting components that would be ideal. My only concern is that formulas with forms code was pretty fragile and painful to maintain in the past, not sure if we want to build more stuff on top of it.

Reply:
The front-end I think was not that bad, only the hide/show part was not obvious (but I think Karl/Richa have work or at least plans to improve it). The main problem is the formulas overall integration